src/adapter/threads.ts

```diff
@@ -868,37 +869,52 @@ export class Thread implements VariableStoreDelegate {
       (source as any)[kScriptsSymbol] = new Set();
     (source as any)[kScriptsSymbol].add(script);
 
-    if (!this._pauseOnSourceMapBreakpointId && event.sourceMapURL) {
-      // If we won't pause before executing this script, try to load source map
-      // and set breakpoints as soon as possible. This is still racy against the script execution,
-      // but better than nothing.
-      this._sourceContainer.waitForSourceMapSources(source).then(sources => {
-        if (sources.length && this._scriptWithSourceMapHandler)
-          this._scriptWithSourceMapHandler(script, sources);
-      });
+    if (event.sourceMapURL) {
+      // If we won't pause before executing this script, still try to load source
+      // map and set breakpoints as soon as possible. This is racy against the
+      // script execution, but better than nothing.
+      this._getOrStartLoadingSourceMaps(script);
     }
   }
 
   // Wait for source map to load and set all breakpoints in this particular script.
   async _handleSourceMapPause(scriptId: string): Promise<boolean> {
     this._pausedForSourceMapScriptId = scriptId;
+    const timeout = this._sourceContainer.sourceMapTimeouts().scriptPaused;
     const script = this._scripts.get(scriptId);
-
-    let remainPaused = false;
-    if (script) {
-      const timeout = this._sourceContainer.sourceMapTimeouts().scriptPaused;
-      const sources = await Promise.race([
-        this._sourceContainer.waitForSourceMapSources(script.source),
-        delay(timeout),
-      ]);
-      if (sources && this._scriptWithSourceMapHandler) {
-        remainPaused = (await this._scriptWithSourceMapHandler(script, sources)).remainPaused;
-      }
+    if (!script) {
+      return false;
```

Comment:
You want to reset `this._pausedForSourceMapScriptId` here.

Reply:
Good catch!